comms/dht/src/network_discovery/discovering.rs

```diff
@@ -149,37 +144,68 @@ impl Discovering {
     }
 
     async fn request_from_peers(&mut self, mut conn: PeerConnection) -> Result<(), NetworkDiscoveryError> {
-        let client = conn.connect_rpc::<rpc::DhtClient>().await?;
+        let rpc_connect_timeout = self.config().network_discovery.bootstrap_rpc_connect_timeout;
+        let rpc_result = tokio::time::timeout(rpc_connect_timeout, conn.connect_rpc::<DhtClient>()).await;
+        let client = match rpc_result {
+            Ok(Ok(client)) => {
+                trace!(
+                    target: LOG_TARGET,
+                    "Discovering: Successfully connected RPC client to sync peer '{}'",
+                    conn.peer_node_id()
+                );
+                client
+            },
+            Ok(Err(e)) => {
+                error!(
+                    target: LOG_TARGET,
+                    "Discovering: Failed to connect RPC client to sync peer {}: {}",
+                    conn.peer_node_id(),
+                    e
+                );
+                return Err(e.into());
+            },
+            Err(_) => {
+                error!(
+                    target: LOG_TARGET,
+                    "Discovering: RPC connect_rpc to sync peer '{}' timed out after {:?}",
+                    conn.peer_node_id(),
+                    rpc_connect_timeout,
+                );
+                return Err(NetworkDiscoveryError::Timeout {
+                    operation: "connect_rpc".to_string(),
+                    peer: conn.peer_node_id().to_hex(),
+                    duration: format!("{:.2?}", rpc_connect_timeout),
+                });
+            },
+        };
         let peer_node_id = conn.peer_node_id();
 
         debug!(
             target: LOG_TARGET,
-            "Established RPC connection to peer `{}`", peer_node_id
+            "Discovering: Established RPC connection to sync peer `{}`", peer_node_id
         );
         let result = self.request_peers(peer_node_id, client).await;
         self.ban_on_offence(peer_node_id.clone(), result).await?;
 
         Ok(())
     }
 
-    async fn request_peers(
+    async fn get_stream(
```

Comment:
```suggestion
    async fn get_peer_stream(
```
for more readable code